src/js/tech/middleware.js

```diff
@@ -0,0 +1,63 @@
+import Tech from './tech.js';
+import toTitleCase from '../utils/to-title-case.js';
+
+const middlewares = {};
+
+export function use(type, middleware) {
+  (middlewares[type] = middlewares[type] || []).push(middleware);
```

Comment:
@mmcc The only problem with `concat` in that case is that if `middlewares[type]` is `undefined`, you'll get an array like `[undefined, middleware]` which I think is probably undesirable.